src/Development/Shake/Internal/Core/Rules.hs

```diff
@@ -147,9 +148,14 @@ runRules opts (Rules r) = do
 --   Returns the command, paired with the documentation (if any).
 getTargets :: ShakeOptions -> Rules () -> IO [(String, Maybe String)]
 getTargets opts rs = do
-    (_actions, _ruleinfo, _userRules, targets) <- runRules opts rs
+    RulesInfo{riTargets = targets} <- runRules opts rs
```

Comment:
I'd have gone with record punning and just written `RulesInfo{riTargets}` and used `riTargets` on the line below. However, its basically personal style.